src/components/VmDetails/cards/DetailsCard.js

```diff
@@ -1,39 +1,258 @@
+    this.setState({ isEditing: true, isDirty: false, correlationId: null, correlatedMessages: null })
+    this.props.onEditChange(true)
+  }
+
+  handleCardOnCancel () {
+    this.setState({ isEditing: false, isDirty: false, correlationId: null, correlatedMessages: null })
+    this.props.onEditChange(false)
+  }
+
+  handleCardOnSave () {
+    const { vm: stateVm } = this.state
+    const correlationId = generateUnique('DetailsCard-save_')
+
+    const vmUpdates = {
+      id: stateVm.get('id'),
+      //
+      // TODO: Select the proper fields to update
+      //
+    }
+    this.setState({ correlationId })
+    this.props.saveChanges(vmUpdates, correlationId)
+
+    return false
+  }
+
+  handleChange (fieldName, { target: { value } }) {
+    if (this.state.isEditing && !this.state.isDirty) {
+      this.props.onEditChange(true, true)
+    }
+
+    const newVm = this.state.vm.set(fieldName, value)
+    this.setState({ vm: newVm, isDirty: true })
+  }
+
+  render () {
+    const { vm, hosts, clusters, dataCenters, templates } = this.props
+    const { isEditing, correlatedMessages } = this.state
+
+    const hostName = (hosts && hosts.getIn([vm.get('hostId'), 'name'])) || '...'
+    const ip4Addresses = !vm.has('nics') ? [] : vm.get('nics').reduce((ipSet, nic) => [...ipSet, ...nic.get('ipv4')], [])
+    const fqdn = vm.get('fqdn') || '?'
+
+    const clusterName = (clusters && clusters.getIn([vm.getIn(['cluster', 'id']), 'name'])) || '...'
+
+    const dataCenterId = (clusters && clusters.getIn([vm.getIn(['cluster', 'id']), 'dataCenterId']))
+    const dataCenter = dataCenters && dataCenters.find(v => v.id === dataCenterId)
+    const dataCenterName = (dataCenter && dataCenter.name) || '...'
+
+    const templateName = (templates && templates.getIn([vm.getIn(['template', 'id']), 'name'])) || '...'
+
+    // look up in __storageDomains__ -- but, need to run 'fetchISOStorages' saga via 'getIsoStorageDomains' redux action to populate the store
+    const canChangeCD = isEditing && vm.get('status') === 'up'
+    const cdImageName = vm.getIn(['cdrom', 'fileId']) || msg.empty()
+    const isoList = createIsoList(this.props.storageDomains) // TODO: Try to make isoList part of state...
+    console.info('isoList', isoList)
+
+    const couldInitEnabled = vm.getIn(['cloudInit', 'enabled'])
+
+    const bootMenuEnabled = vm.get('bootMenuEnabled')
+
+    return (
+      <BaseCard
+        title='Details'
+        editMode={isEditing}
+        editTooltip={`Edit details for ${vm.get('id')}`}
+        onStartEdit={this.handleCardOnStartEdit}
+        onCancel={this.handleCardOnCancel}
+        onSave={this.handleCardOnSave}
+      >
+        {({ isEditing }) => {
+          return (<React.Fragment>
+            <Grid className={style['details-container']}>
+              <Row>
+                <Col className={style['fields-column']}>
+                  <Grid>
+                    <FieldRow label={'Status'}>...status...</FieldRow>
```

Comment:
Translate please. And at the bottom as well.

Reply:
I left lots of things un-translated.  Will need to clean this up in later PRs.